lib/m3u8-parser.js

```diff
@@ -34,14 +37,33 @@ module.exports = class m3u8parser extends Writable {
   _write(chunk, encoding, callback) {
     var lines = chunk.toString('utf8').split('\n');
     if (this._lastLine) { lines[0] = this._lastLine + lines[0]; }
-    lines.forEach((line, i) => {
-      if (i < lines.length - 1) {
-        this._parseLine(line);
-      } else {
-        // Save the last line in case it has been broken up.
-        this._lastLine = line;
+    if (!this.options.youtubeFormatting) {
+      lines.forEach((line, i) => {
+        if (i < lines.length - 1) {
+          this._parseLine(line);
+        } else {
+          // Save the last line in case it has been broken up.
+          this._lastLine = line;
+        }
+      });
+    } else {
+      this._lastLine = lines[lines.length - 2];
```

Comment:
Set the last line to before the blank output [' '], which is two back. This is the last audio output.